feat: add support for custom projects (#802)
open-energy-transition/open-tyndp: doc/cba.md

$ pixi run tyndp-cba --config run='{"name":"NT-cy2009"}'
```

## Evaluation of custom projects

> Custom PINT transmission projects can be evaluated with the CBA workflow. Each project is defined in `data/custom_cba_transmission_projects.csv` and selected in the configuration. Two types of custom project are supported, depending on whether `project_id` refers to an existing PINT project:

Reviewer: ```suggestion
Custom transmission projects can be evaluated with the CBA workflow. Each project is defined in `data/custom_cba_transmission_projects.csv` and selected in the configuration. Two types of custom project are supported, depending on whether `project_id` refers to an existing project:
```

Author: Decided to focus on PINT for now

> - **Modified projects**: if the combination (`project_id`, `bus0`, `bus1`) matches an existing PINT project, the specified fields overwrite those of that project. Fields left empty retain their original values.

Reviewer: ```suggestion
- **Modified projects**: if the combination (`project_id`, `bus0`, `bus1`) matches an existing project, the specified fields overwrite those of that project. Fields left empty retain their original values.
```

Author: Decided to focus on PINT for now

- **New projects**: if `project_id` is not yet used, the entry is added as a new PINT project, modeled as a link.

> Every entry must define `project_id`, `bus0`, `bus1` and at least one capacity (`p_nom 0->1` or `p_nom 1->0`), and the resulting combinations must be unique. Entries referring to TOOT projects are not supported and are ignored with a warning.

Reviewer: ```suggestion
Every entry must define `project_id`, `bus0`, `bus1` and at least one capacity (`p_nom 0->1` or `p_nom 1->0`), and the resulting combinations must be unique. For entries referring to TOOT projects the user has to make sure that the project is part of the reference grid.
```

Author: Decided to focus on PINT for now
